build.cake

```
@@ -103,7 +105,17 @@ private string GetCommitCount()
     return redirectedOutput.Last();
 }
 
-private string GetVersionNumberFromTag()
+private string GetFormattedTimestamp()
+{
+    var now = DateTime.UtcNow;
+    // we append a random value greater than 14 to the version number to not clash with the versions generated from tags
+    // the last component in the tag generated version is the revision number, which so far has never been more than 5 
+    var randomComponent = 15 + rnd.Next(85);
```

Maybe it would make sense to define 15 as a constant with some proper name like `MAXIMUM_REVISION_NUMBER`? We can also use this constant in tag-based version generation and let it crash or at least log some warning that you are exceeding the maximum supported revision number.  You can also define the `rnd.Next(85);` as `rnd.Next(100-MAXIMUM_REVISION_NUMBER);` so it's always in sync in case we need to change `MAXIMUM_REVISION_NUMBER` in the future.